### ISSUE TITLE
Feature requests for Springboot 

### ISSUE THREAD

**Issue by texneus** [Libresonic/libresonic#283]
It is possible that some? all? of these features may already be present but due to the current state of documentation and my lack of knowledge about Springboot I just cannot implement them.

- I was able to launch the 6.1 Beta 1 .WAR in FreeBSD 11 without issue, although this page suggests a custom embeddedLaunchScript may be required for FreeBSD and OSX -- http://docs.spring.io/spring-boot/docs/current/reference/html/deployment-install.html.  Requirements for Windows are not mentioned.  All that I am really suggesting for now is the minimal install document be updated to reflect what OS's the Springbooted .WAR was designed to operate on.  I do understand documentation is probably low priority at the moment, but I think if people know it is available and is supposed to work on their platform you might get more feedback on it's effectiveness.  Setting up a Subsonic .WAR and updates was always kind of a pain...I would consider this revolutionary!

- Please provide the option in the Springboot .WAR to use HTTPS connection and certificates or some other encryption technology.  I would rather not have my user credentials passed in plain text.  Also, since Libresonic can be used for the transfer of copyrighted materials I strongly suggest a mechanism to protect content in transmission be a firm requirement.

- "Secure" the Springboot embedded Tomcat, if possible.  For example, it was rather trivial for me to confirm the server is Tomcat and also to find which version.  It is my opinion that just knowing what server and version is operating behind the scenes translates directly into what hacks are available to use against it.  There are many guides in how to do this, most are simple edits to the Tomcat configuration files.  For example:  https://www.upguard.com/articles/15-ways-to-secure-apache-tomcat-8, http://security-24-7.com/hardening-guide-for-tomcat-8-on-redhat-6-5-64bit-edition/

_edit to add:_
- Run Libresonic to run as a non-root process.  At this time I get the "running as root" warning but it isn't clear to me how to run it any differently.  Presumably a non-root user needs to be setup and provided (somehow) to Librisonic at launch.

**Comment by jooola** [comment]
* For 2 : We recommend using the ssl encryption, by using a reverse proxy in front of the libresonic backend. Here is the current [PROXY documentation](https://github.com/Libresonic/libresonic/blob/develop/documentation/PROXY.md).
* For 3 : This is a good idea ! I guess we could add this to the documentation !

**Comment by texneus** [comment]
Why not use the TLS capabilities built in to Tomcat?  I've been operating it this way for a long time and 6.1 beta 1 (under Tomcat 8, not springbooted) seems to operate properly even though I think I read SSL has already been removed.  Or am I misunderstanding how https in Tomcat works and SSL is still in Libresonic?  I can send instructions on how I'm setting this up if it helps.

**Comment by jooola** [comment]
I dont know how tomcat SSL implementation works , but i know that we removed the integrated SSL feature from libresonic because we think a reverse proxy is way more powerfull !
If you get a tomcat with SSL working i guess we will take any help for the documentation.

**Comment by texneus** [comment]
I haven't pushed it real hard but it definitely seems to work.  I admit I'm not an expert on this stuff (I know just enough to be dangerous with this stuff), but I did go review the proxy directions and why I might want one.  I now see why a person might want this, but for a single purpose server that only points Libresonic to the web, it seems like an extra layer of stuff to keep up to date, maintain, or to just break.  Seems like Tomcat's built in HTTPS might still be a better option for *some*.

Anyway I'm attaching a document that I wrote as I installed Libresonic, secured Tomcat, and enabled HTTPS via a self signed certificate in FreeBSD.  This is not valid for a Springbooted .WAR as written, but presumably the steps can be modified to do so.  Perhaps all this is really only necessary if there is no proxy?  Anyway, feel free to use this as you see fit...

Doh!  GitHub doesn't support LibreOffice documents, not even if I save it as an MS-Office DOCX that is supposed to be accepted????

[Libresonic - Tomcat.pdf](https://github.com/Libresonic/libresonic/files/839963/Libresonic.-.Tomcat.pdf)

**Comment by muff1nman** [comment]
Excellent doc! I imagine this will help out some that want to go the Secure Tomcat route. As you are probably aware the setup is quite intense. IMO the proxy route is a bit simpler and requires less meddling with things (i.e. you know you're deep when you're repacking jars) even though it requires two processes running. But this is definitely a valid route to take to SSL.

I think maybe to close this issue, we can just decide where to put that doc - I think its worth putting into /documentation even though it isn't markdown format.

**Comment by texneus** [comment]
I originally created this to consider adding these features to the Tombat embedded in Springboot.  Not having to install Tomcat is already a big advantage (just run and done!), but not having to go through this ritual would be the icing on the cake.  Since the directions I provided would preserve HTTP, those who opt for a proxy should still be able to do so.  Proxy or not, everyone would get to take advantage of a Tomcat that is more secure than it is "out of the box"

If it's not possible to do this inside the Springboot, then directions in the final documentation to do it manually would be sufficient.  If it's not possible to do at all then the "old fashioned" way as I presented above is fine too.

I'd be happy to revisit this doc if you like for those who want to run it stand alone with HTTPS (I noticed a few typos that need to be fixed anyway).  Options are always a good thing!  I could also add decoder setup (etc) so it's truely a start to end document.   I have visitors this week so I won't be able to loo do anything for a little while and I don't speak Markdown though...
